### DISCUSSION TITLE
How to create ifc structure tree (c++)

### DISCUSSION THREAD

**Discussion by ShinjiOhnishi** [IfcOpenShell/IfcOpenShell#4400] [Q&A]
I want to create ifc structure tree like "IFC Structure" in BIMVision using IFCOpenShell v0.7.0.
I found following example but it doesn't work with v0.7.0.
https://gist.github.com/stefkeB/e2b308e13d299328ecd1dc8db2d20cc5
Is there any code example or documentation help implementing?

**Comment by theoryshaw** [comment]
maybe this? https://github.com/OpeningDesign/ifc-commenter

**Comment by Moult** [comment]
Have you got a screenshot of what you're after?

**Comment by aothms** [comment]
The code is very old, but here is also a quick version of this without dependencies https://github.com/IfcOpenShell/IfcOpenShell/blob/b0156475436c99628fd8e95d3163bc72e6f0da90/src/ifcopenshell-python/ifcopenshell/geom/app.py#L261
